### PR TITLE
[REEF-1709] Make unit tests in reef-runtime-yarn compile and work again

### PR DESCRIPTION
**Pull request by motus**
Move unit tests into a proper directory in maven directory structure and refactor the code to get rid of checkstyle errors (indentation, long lines)

JIRA: [REEF-1709](https://issues.apache.org/jira/browse/REEF-1709)

Closes PR #

### CODE REVIEW

File: lang/java/reef-runtime-yarn/src/test/java/org/apache/reef/runtime/yarn/driver/YarnResourceRequestHandlerTest.java [hunk 1]
```diff
@@ -57,70 +61,75 @@ public void onContainerRequest(AMRMClient.ContainerRequest... containerRequests)
...
+        requestOne.getMegaBytes(),
+        containerRequestHandler.getRequests()[0].getCapability().getMemory());
+
+    evaluatorRequestor.submit(requestTwo);
+    Assert.assertEquals("Request in REEF and YARN form should have the same amount of memory",
+        requestTwo.getMegaBytes(),
+        containerRequestHandler.getRequests()[0].getCapability().getMemory());
+
+    evaluatorRequestor.submit(requestOne);
+    Assert.assertNotEquals("Request in REEF and YARN form should have the same amount of memory",
```
Comment (tcNickolas) [inline lang/java/reef-runtime-yarn/src/test/java/org/apache/reef/runtime/yarn/driver/YarnResourceRequestHandlerTest.java]: Can you please update the message in `assertNotEquals` to reflect what it actually checks?

File: lang/java/reef-runtime-yarn/src/test/java/org/apache/reef/runtime/yarn/driver/YarnResourceRequestHandlerTest.java [hunk 2]
```diff
@@ -57,70 +61,75 @@ public void onContainerRequest(AMRMClient.ContainerRequest... containerRequests)
...
+        requestOne.getNumber(),
+        containerRequestHandler.getRequests().length);
+
+    evaluatorRequestor.submit(requestTwo);
+    Assert.assertEquals("Request in REEF and YARN form should have the same number of Evaluators",
+        requestTwo.getNumber(),
+        containerRequestHandler.getRequests().length);
+
+    evaluatorRequestor.submit(requestTwo);
+    Assert.assertNotEquals("Request in REEF and YARN form should have the same number of Evaluators",
```
Comment (tcNickolas) [inline lang/java/reef-runtime-yarn/src/test/java/org/apache/reef/runtime/yarn/driver/YarnResourceRequestHandlerTest.java]: Can you please update the message in assertNotEquals to reflect what it actually checks?

**Comment by tcNickolas** [comment]
Judging from the names, these tests are supposed to run on Yarn. Is this correct? Have you tested this change on Yarn? Travis only runs local tests :-(

**Comment by motus** [comment]
@tcNickolas thanks for reviewing! I've updated the assertion messages. As for the YARN tests, these specific tests do not require access to the actual cluster and should pass on our CI servers.

**Pull request closed without merging**